### PR TITLE
fix: Allow empty array dimensions

### PR DESCRIPTION
**Pull request by dbrattli**
- Add the proposed fix for https://cognitedata.atlassian.net/browse/CDFBUG-370
- Update README with .NET 8

### CODE REVIEW

File: Extractor/Nodes/UADataType.cs [hunk 1]
```diff
@@ -238,7 +238,8 @@ public override string ToString()
                     return false;
                 }
             }
-            else if (node.ArrayDimensions == null)
+            // Check for null ArrayDimensions or empty array
+            else if (node.ArrayDimensions == null || node.ArrayDimensions.Length == 0)
             {
                 if (config.UnknownAsScalar && (node.ValueRank == ValueRanks.ScalarOrOneDimension
                     || node.ValueRank == ValueRanks.Any)) return true;
```
Comment (kornelione) [inline Extractor/Nodes/UADataType.cs:245]:
```suggestion
                    || node.ValueRank == ValueRanks.Any)) 
                {
                    return true;
                }
```

**Review by gregertw — APPROVED** [review]
Per incident INC-272, approving as incident lead

**Review by kornelione — APPROVED** [review]

**Review by einarmo — APPROVED** [review]
Looks like a reasonable change.

**Comment by dbrattli** [comment]
Do not merge before we have updated the manifest

**Pull request merged**